### PR TITLE
feat(ui): persist oidc auth details

### PR DESCRIPTION
**Pull request by jayeshchoudhary**
This PR is `ui` `feature`

### Issue
- Users need to log in every time the app is refreshed because the authentication token is stored locally in the app, making it inaccessible after a refresh.
- The app currently redirects users to the home page after login, rather than to the specific page they intended to visit before authentication.

### Solution
- UI should persist your authentication details in the browser session. This means that even if you refresh the app, you will still be logged in until the authentication session expires.
- Redirect to the appropriate location after login, so you will be taken directly to the page you intended to visit before authentication.

### Note
The solution is only implemented for OIDC authentication. Other auth methods will work as before.

Reviewers - @apucher @joshigaurava @mayankshriv

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
cc @joshigaurava @apucher

File: pinot-controller/src/main/resources/app/App.tsx [hunk 1]
```diff
@@ -18,48 +18,42 @@
...
-  const [clientId, setClientId] = React.useState(null);
-  const [authWorkflow, setAuthWorkflow] = React.useState(null);
-  const [authorizationEndpoint, setAuthorizationEndpoint] = React.useState(
-    null
-  );
   const [role, setRole] = React.useState('');
+  const { authenticated, authWorkflow } = useAuthProvider();
+  const history = useHistory();
+
+  React.useEffect(() => {
```
Comment (joshigaurava) [inline pinot-controller/src/main/resources/app/App.tsx:38]: Minor and not a blocker: can you pleease fix the import so that you can use just `useEffect`?

Comment (jayeshchoudhary) [reply pinot-controller/src/main/resources/app/App.tsx]: fixed.

**Comment by joshigaurava** [comment]
Looks good.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**